  Hello,

> On Wed, 22 Aug 2001, Jan Kara wrote:
> 
> >   Hello,
> > 
> >   Jan Sanislo <oystr@cs.washington.edu> found a bug in ext2 quota code in 2.4.6+.
> > During changes in ext2 code in 2.4.6 some DQUOT_INIT()s were removed but they
> > shouldn't and as a result quota is not computed right.
> >   The patch which adds missing DQUOT_INIT()s is below. I didn't place DQUOT_INIT()s to
> > original places but rather to generic vfs parts which seems better to me.
> >   Please apply - patch is against 2.4.8.
> 
> Wrong place - DQUOT_INIT should be in iput(), just before the call of
> ->delete_inode()
  OK. I was also thinking about this place but finally I chose those places in namei.c..
bad choice :)
  The patch which places DQUOT_INIT() in iput() is below. Please apply.

										Honza
--
Jan Kara <jack@suse.cz>
SuSE Labs
------------------------------------------------------------------------------------

--- linux-2.4.9/fs/inode.c	Wed Aug 22 17:00:51 2001
+++ linux-2.4.9/fs/inode.c	Wed Aug 22 17:01:11 2001
@@ -1049,6 +1049,7 @@
 
 			if (op && op->delete_inode) {
 				void (*delete)(struct inode *) = op->delete_inode;
+				DQUOT_INIT(inode);
 				/* s_op->delete_inode internally recalls clear_inode() */
 				delete(inode);
 			} else
-
To unsubscribe from this list: send the line "unsubscribe linux-kernel" in
the body of a message to majordomo@vger.kernel.org
More majordomo info at  http://vger.kernel.org/majordomo-info.html
Please read the FAQ at  http://www.tux.org/lkml/
